### PR TITLE
Build via Jenkins pipeline

### PR DESCRIPTION
**Pull request by schniedergers**
This adds a Jenkins-run pipeline and docker support.

The Jenkins pipeline will be run in parallel to Travis, for all PRs, tags, and pushes to master.
Test results are converted and parsed, see https://jenkins.ci.swift-nav.com/job/swift-nav/job/libswiftnav/job/PR-43/4/testReport/(root)/ as an example.

Also added a simple top-level Makefile for running builds in docker form a dev environment. Instructions are in the README.md.
Basically, just do "make docker-build" after being logged into AWS, and it should download/build/start a container and run cmake/make in it.

Since this uses a ECR-hosted docker base image, these will only work for swift users/employees.

### CODE REVIEW

**Comment by schniedergers** [comment]
(goes together with https://github.com/swift-nav/ci-jenkins/pull/39 which actually enables the CI pipeline for this repo)

**Comment by schniedergers** [comment]
Do not merge - I'll restructure the docker image setup to mirror what's proposed for starling in https://github.com/swift-nav/starling/pull/2020
